test/format/ops.c

```diff
@@ -1395,6 +1400,23 @@ ops(void *arg)
             break;
         }
 
+        if (mirrored_truncate) {
+            /*
+             * At the end of a mirrored truncate all tables must contain the same keys. It's ok if a
+             * parallel insert has added keys back inside the truncated range as long as all mirror
+             * tables have that same key.
+             *
+             * Verifies can be expensive so we limit them to smaller ranges and only infrequently
+             * check larger ranges.
+             */
+            if (tinfo->last != 0 && tinfo->keyno != 0) {
+                if ((tinfo->last - tinfo->keyno) < 10000)
```

Comment:
Verifying 100% of truncates shorter than 10,000 keys and 10% of truncates larger than 10,000 keys was based on inspection of local test/format runs. I'm happy to discuss better numbers here

Reply:
I like this approach. Do you think we should keep this indefinitely though?
One more thing, if we don't see the start or the end of the range, we should still be able to know how many keys we are deleting.

Reply:
Fair question.. I like keeping this long term as otherwise we'll only catch mirrored truncate errors at checkpoint time like in WT-11507, and by that point we've lost the relevant information.
How about we review this at the same time we review the failure config we added? If it hasn't fired by then we can remove it.

> One more thing, if we don't see the start or the end of the range, we should still be able to know how many keys we are deleting.

I wanted to avoid verifying open ended truncates (`if (tinfo->last != 0 && tinfo->keyno != 0)`) since they're likely to be very expensive and slow. Do you mean you'd want to also verify these open ended truncates with code along the lines of
```
if(tinfo->last == 0)
    truncate_begin = 1;
if(tinfo->keyno == 0)
    truncate_end = TV(RUNS_ROWS);

if((truncate_end - truncate_begin) > 1000)
    // existing code
```
?

Reply:
I am wondering if the beginning is not set but the end is, we still can determine if the range is > N since the beginning should be 0.
Same if the last is not set but the start is, we should be able to know how many keys are covered.
Finally, if both are unset, we should check how many keys the table contains to know whether the check should be done or not.

Reply:
Updated in [d9fd6a8](https://github.com/wiredtiger/wiredtiger/pull/9660/commits/d9fd6a86fafaf9449bed957e42c84ccd3e114973). I've pulled the logic into a separate `wts_verify_mirrored_truncate` function for convenience.